doc/source/toolchain.rst

```diff
@@ -0,0 +1,226 @@
+Toolchain Roadmap
+=================
+
+The use of the SciPy library requires (or optionally depends upon) several
+other libraries in order to operate, the main dependency being Python and NumPy.
+It requires a larger collection of libraries and tools in order to build the library,
+or to build the documentation.
+
+Of course, the tooling and libraries are themselves not static.  This document aims to provide a
+guide as to how SciPy's use of these dynamic dependencies will proceed over time.
+
+SciPy aims to be compatible with a number of releases of its dependent libraries and tools.
+Forcing the user base to other components for upgrade for every release would greatly
+diminish the value of SciPy.  However, maintaining backwards compatibility
+with very old tooling/libraries imposes limitations on which newer functionalities and capabilities
+can be incorporated.  SciPy takes a somewhat conservative approach, maintaining compatibility with
+several major releases of Python and NumPy on the major platforms. (That may in of itself impose
+further restrictions.  See the C Compilers section for an example.)
+
+
+- First and foremost, SciPy is a Python project hence it requires a Python environment.
+- The LAPACK and OpenBLAS numerical libraries need to be installed.
+- Compilers for C, C++, Cython and Fortran code are needed.
+- The Python environment needs the ``NumPy`` package to be installed.
+- Testing requires the ``pytest`` Python package.
+- Building the documentation requires the ``matplotlib``, Sphinx and ``numpydoc`` packages, as well as a LaTeX installation.
+
+The tooling used to build CPython has some implications for the tooling used in building SciPy.
+It also has implications for the documentation in docstrings.
```

Comment:
I'm not sure what the implications for docstrings is referring to?

Reply:
Example code in the docstrings has to be executable with a range of Python/NumPy version.
Updated the text to be clearer.